experimental/fluent/func/src/main/java/io/serverlessworkflow/fluent/func/dsl/UniqueIdBiFunction.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2020-Present The Serverless Workflow Specification Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.serverlessworkflow.fluent.func.dsl;
+
+/**
+ * Functions that expect a unique ID injection in runtime, typically an idempotent generated unique
+ * id based on the workflow instance id and task name.
+ *
+ * @param <T> The task payload input
+ * @param <R> The task result output
+ */
+@FunctionalInterface
+public interface UniqueIdBiFunction<T, R> {
```

Comment:
```suggestion
public interface UniqueIdBiFunction extends BiFunction<WorkflowContext,TaskContext,String> {
```

And the implementations in the PR should be changed accordingly, We can have a method the generates the default uniqueId, which implementation should be

```
public static String defaultUniqueId (WorkflowContext wctx, TaskContext tctx) {
 return String.format("%s-%s", wctx.instanceData().id(), tctx.position().jsonPointer());
}

```

Reply:
I agree about the static method to document the API, but `UniqueIdBiFunction` comes from clients:

```java
  public static <T, R> FuncCallStep<T, R> agent(
      String name, UniqueIdBiFunction<T, R> fn, Class<T> in) {
    return withUniqueId(name, fn, in);
  }
```

Their methods supposed to be like:

```java
function chat(@MemoryId String id, @V String message);
```

There's no internal workflow domain.